### PR TITLE
Change Reject To Error Out When Invalid Column is Provided

### PR DESCRIPTION
**Pull request by JosephTLyons**
Currently, in both `reject` and `select`, you can provide column names that don't exist and nothing happens.  This was a bit confusing for me at one point when I went to `select` some longer column name that I misspelled, but it wasn't obvious to me at first, and nothing was happening.  This PR causes `reject` to error out if trying to reject a column that isn't there.  A PR for `select` will follow.

```shell
> ls | reject name
────┬──────┬──────────┬───────────────
 #  │ type │   size   │   modified    
────┼──────┼──────────┼───────────────
  0 │ File │   3.4 KB │ 2 days ago    
  1 │ File │   1.7 KB │ 18 hours ago  
  2 │ File │ 145.8 KB │ 18 hours ago  
  3 │ File │   6.8 KB │ 18 hours ago  
  4 │ File │   1.1 KB │ 3 months ago  
  5 │ File │    449 B │ 3 months ago  
  6 │ File │    192 B │ 2 days ago    
  7 │ File │  15.9 KB │ 18 hours ago  
  8 │ Dir  │    928 B │ 18 hours ago  
  9 │ Dir  │    352 B │ 3 months ago  
 10 │ Dir  │    288 B │ 18 hours ago  
 11 │ Dir  │    256 B │ 18 hours ago  
 12 │ Dir  │    160 B │ 3 months ago  
 13 │ Dir  │     96 B │ 18 hours ago  
 14 │ File │     16 B │ 10 months ago 
 15 │ Dir  │     96 B │ 18 hours ago  
 16 │ Dir  │    128 B │ 18 hours ago  
 17 │ Dir  │    192 B │ 2 weeks ago   
 18 │ Dir  │    192 B │ 3 months ago  
 19 │ Dir  │    128 B │ 18 hours ago  
────┴──────┴──────────┴───────────────
> ls | reject name dog
error: Invalid column
  ┌─ shell:1:18
  │
1 │ ls | reject name dog
  │                  ^^^ invalid column
```

### CODE REVIEW

File: crates/nu-data/src/base.rs [hunk 1]
```diff
@@ -64,20 +64,34 @@ pub fn select_fields(obj: &Value, fields: &[String], tag: impl Into<Tag>) -> Val
...
+    tag: impl Into<Tag>,
+) -> Result<Value, ShellError> {
     let mut out = TaggedDictBuilder::new(tag);
 
     let descs = obj.data_descriptors();
 
+    for field in fields {
+        if descs.iter().all(|desc| desc != &field.item) {
+            return Err(ShellError::labeled_error(
+                "Invalid column",
```
Comment (JosephTLyons) [inline crates/nu-data/src/base.rs:79]: The original for loop below this does not give access to each field, which we need to grab the tag for the error, so I had to basically those original nested loops and flip them inside out in order to get the check to work.

File: crates/nu-data/src/base.rs [hunk 2]
```diff
@@ -64,20 +64,34 @@ pub fn select_fields(obj: &Value, fields: &[String], tag: impl Into<Tag>) -> Val
...
+                "invalid column",
+                field.tag.clone(),
+            ));
+        }
+    }
+
```
Comment (JosephTLyons) [inline crates/nu-data/src/base.rs:85]: The original for loop below this does not give access to each field, which we need to grab the tag for the error, so I had to basically those original nested loops and flip them inside out in order to get the check to work.

Comment (jonathandturner) [reply crates/nu-data/src/base.rs:85]: I was wondering about that. we want to check that at least one of the rows has a column with that name, right? If some don't but others do, seems like that'd still be okay.

Comment (JosephTLyons) [reply crates/nu-data/src/base.rs:85]: Hmm, where do those situations come up?

```shell
> open Cargo.toml | get bin | flatten required-features | last 3
───┬─────────────────────────────┬────────────────────────────────────────────┬───────────────────
 # │            name             │                    path                    │ required-features 
───┼─────────────────────────────┼────────────────────────────────────────────┼───────────────────
 0 │ nu_plugin_extra_from_sqlite │ src/plugins/nu_plugin_extra_from_sqlite.rs │ sqlite            
 1 │ nu_plugin_extra_to_sqlite   │ src/plugins/nu_plugin_extra_to_sqlite.rs   │ sqlite            
───┴─────────────────────────────┴────────────────────────────────────────────┴───────────────────
───┬──────┬─────────────
 # │ name │    path     
───┼──────┼─────────────
 2 │ nu   │ src/main.rs 
───┴──────┴─────────────
> open Cargo.toml | get bin | flatten required-features | last 3 | reject required-features
───┬─────────────────────────────┬────────────────────────────────────────────
 # │            name             │                    path                    
───┼─────────────────────────────┼────────────────────────────────────────────
 0 │ nu_plugin_extra_from_sqlite │ src/plugins/nu_plugin_extra_from_sqlite.rs 
 1 │ nu_plugin_extra_to_sqlite   │ src/plugins/nu_plugin_extra_to_sqlite.rs   
───┴─────────────────────────────┴────────────────────────────────────────────
error: Invalid column
  ┌─ shell:1:73
  │
1 │ open Cargo.toml | get bin | flatten required-features | last 3 | reject required-features
  │                                                                         ^^^^^^^^^^^^^^^^^ invalid column 
```

Comment (JosephTLyons) [reply crates/nu-data/src/base.rs:85]: I guess I'm not really sure how to deal with that situation.

Comment (jonathandturner) [reply crates/nu-data/src/base.rs:85]: Yeah exactly. When you have a mix of rows that have different shapes. For these I think we should detect if we successfully removed the column at least once. If we get to the end and never did, we should give the error then.

Comment (JosephTLyons) [reply crates/nu-data/src/base.rs:85]: Would that imply that reject and select would no longer be able to stream their results?  As we would have to evaluate all the rows in order to know that at least one row has the column?

Comment (jonathandturner) [reply crates/nu-data/src/base.rs:85]: They can still stream. It's just that at the end we'd also send an error. For some cases the user would never know, like:

```
ls | select nammme
```

Would give no results and error at the end of the stream. If there's no end then there still *may* be a row that matches, so we keep looking. Most people aren't doing infinite streams, so they should fall into the happy case.

Comment (JosephTLyons) [reply crates/nu-data/src/base.rs:85]: Sorry for the delay here, I do plan to figure out a way to do this, will probably do it over the weekend.

**Comment by JosephTLyons** [comment]
Closing this for now, am having a difficult time figuring out how to do what has been suggested.

**Pull request closed without merging**